test/wdio/slot-fallback/cmp.test.tsx

```diff
@@ -0,0 +1,131 @@
+import { h } from '@stencil/core';
+import { render } from '@wdio/browser-runner/stencil';
+
+describe('slot-fallback', () => {
+  beforeEach(() => {
+    render({
+      template: () => <slot-fallback-root></slot-fallback-root>,
+    });
+  });
+
+  it('renders fallback', async () => {
+    // show fallback content
+    await expect($('.results1 slot-fb[name="start"]:not([hidden])')).toHaveText('slot start fallback 0');
+    await expect($('.results1 section slot-fb:not([hidden])')).toHaveText('slot default fallback 0');
+    await expect($('.results1 article span slot-fb[name="end"]:not([hidden])')).toHaveText('slot end fallback 0');
+
+    // update fallback content
+    await $('button.change-fallback-content').click();
+    await browser.pause();
+
+    await expect($('.results1 slot-fb[name="start"]:not([hidden])')).toHaveText('slot start fallback 1');
+    await expect($('.results1 section slot-fb:not([hidden])')).toHaveText('slot default fallback 1');
+    await expect($('.results1 article span slot-fb[name="end"]:not([hidden])')).toHaveText('slot end fallback 1');
+
+    // set light dom instead and hide fallback content
+    await $('button.change-light-dom').click();
+    await browser.pause();
+
+    // fallback content hidden but still the same
+    // webdriverio seems to have some trouble with the `[hidden]` selector, which is why we don't use it here
```

Comment:
Making the comment more clear. WebDriver's `getText` command is defined to only fetch the `visible` text from an element
```suggestion
    // WebdriverIO's `getText` command can only fetch a visible text, which is why we don't use it here
```